main.lua

```diff
@@ -0,0 +1,66 @@
+local lib = require "luaclang"
+local inspect = require "inspect"
+
+--table containing all declarations in the file
+local declarations = {}
```

Comment:
Usually people names this table the name of the module. In your case, you could name it cparser.
```suggestion
local cparser = {}
```
Also, rename the file to something like cparser.lua.
Another thing: I would remove the comment in the previous line.